src/test/java/com/google/ehub/servlets/ProfileServletTest.java

```diff
@@ -149,6 +149,57 @@ public void postResquestWithFalseEditParam_SendsRedirect() throws IOException {
     verify(response).sendRedirect(REDIRECT);
   }
 
+  @Test
+  public void postResquestWithTrueEditParam_SendsRedirect() throws IOException {
+    Entity userEntity = new Entity(PROFILE_ITEM_KIND);
+
+    userEntity.setProperty(NAME_PROPERTY_KEY, NAME);
+    userEntity.setProperty(EMAIL_PROPERTY_KEY, EMAIL);
+    userEntity.setProperty(USERNAME_PROPERTY_KEY, USERNAME);
+    userEntity.setProperty(BIO_PROPERTY_KEY, BIO);
+
+    datastoreService.put(userEntity);
+
+    when(request.getParameter(NAME_PROPERTY_KEY)).thenReturn(NAME);
+    when(request.getParameter(EMAIL_PROPERTY_KEY)).thenReturn(EMAIL);
+    when(request.getParameter(BIO_PROPERTY_KEY)).thenReturn(BIO);
+    when(request.getParameter(USERNAME_PROPERTY_KEY)).thenReturn(USERNAME);
+    when(request.getParameter(EDIT_PROPERTY_KEY)).thenReturn("true");
+
+    servlet.doPost(request, response);
+
+    verify(response).sendRedirect(REDIRECT);
+  }
+
+  @Test
+  public void postResquestWithSameUsername_ErrorGetsSent() throws IOException {
+    Entity userEntity = new Entity(PROFILE_ITEM_KIND);
+    Entity sameEntity = new Entity(PROFILE_ITEM_KIND);
+
+    userEntity.setProperty(NAME_PROPERTY_KEY, NAME);
+    userEntity.setProperty(EMAIL_PROPERTY_KEY, EMAIL);
+    userEntity.setProperty(USERNAME_PROPERTY_KEY, USERNAME);
+    userEntity.setProperty(BIO_PROPERTY_KEY, BIO);
+
+    sameEntity.setProperty(NAME_PROPERTY_KEY, NAME);
+    sameEntity.setProperty(EMAIL_PROPERTY_KEY,"os@gmail.com");
+    sameEntity.setProperty(USERNAME_PROPERTY_KEY, "os");
+    sameEntity.setProperty(BIO_PROPERTY_KEY, BIO);
+
+    datastoreService.put(userEntity);
+    datastoreService.put(sameEntity);
```

Comment:
You can make a separate function to create an entity and add it to datastore to prevent code duplication.

Reply:
The reason why there is duplicate code is because I needed to create another entity that already has a username. So that  it can test when a username is being changed to one that already exists.